### PR TITLE
[Feature][CheckStyle] Add Scala CheckStyle support

### LINKED ISSUE #863: [Feature][CheckStyle] Add Scala CheckStyle support
### Search before asking

- [X] I had searched in the [feature](https://github.com/InterestingLab/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Add Scala CheckStyle support

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by wuchunfu**
## Purpose of this pull request

close #863

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] Change does not need document change, or I will submit document change to https://github.com/InterestingLab/seatunnel-docs later

### CODE REVIEW

**Comment by CalvinKirs** [comment]
Can you try to join CI Check?

**Comment by wuchunfu** [comment]
> Can you try to join CI Check?

Ok, let me add.

File: scalastyle-config.xml [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+  ~
+-->
+
+<scalastyle>
+    <name>Scalastyle standard configuration</name>
+
+    <check level="error" class="org.scalastyle.file.HeaderMatchesChecker" enabled="true">
+        <parameters>
+            <parameter name="header"><![CDATA[/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
```
Comment (CalvinKirs) [inline scalastyle-config.xml:26]: We have used skywalking-eyes to check

**Comment by CalvinKirs** [comment]
Your pr has other people's submissions, please rebase your code correctly.

File: scalastyle-config.xml (hunk 1 above)
Comment (wuchunfu) [reply scalastyle-config.xml:26]: Okay, I will delete this part of the content

**Comment by wuchunfu** [comment]
> Your pr has other people's submissions, please rebase your code correctly.

Okay

**Comment by CalvinKirs** [comment]
`/tools/checkstyle` Would it be better to put it in this directory?

**Comment by wuchunfu** [comment]
> /tools/checkstyle Would it be better to put it in this directory?

Ok, let me adjust

**Comment by wuchunfu** [comment]
> > `/tools/checkstyle` Would it be better to put it in this directory?
> 
> Ok, let me adjust

Adjusted cc @CalvinKirs

**Review by CalvinKirs — APPROVED** [review]
Good Job~

**Pull request merged**